nodebuilder/state/core.go

```diff
@@ -18,21 +10,9 @@ import (
 // CoreAccessor constructs a new instance of state.Module over
 // a celestia-core connection.
 func CoreAccessor(
-	lc fx.Lifecycle,
-	fService fraudServ.Module,
 	corecfg core.Config,
 	signer *apptypes.KeyringSigner,
 	sync *sync.Syncer,
-) Module {
-	service := state.NewCoreAccessor(signer, sync, corecfg.IP, corecfg.RPCPort, corecfg.GRPCPort)
-	lc.Append(fx.Hook{
-		OnStart: func(ctx context.Context) error {
-			lifecycleCtx := fxutil.WithLifecycle(ctx, lc)
-			return fraudServ.Lifecycle(ctx, lifecycleCtx, fraud.BadEncoding, fService, service.Start, service.Stop)
-		},
-		OnStop: func(ctx context.Context) error {
-			return service.Stop(ctx)
-		},
-	})
-	return service
+) *state.CoreAccessor {
+	return state.NewCoreAccessor(signer, sync, corecfg.IP, corecfg.RPCPort, corecfg.GRPCPort)
```

Comment:
Can you use the constructor directly as an fx.Provide?

Reply:
No, because we need to unpack the config values into the constructor - there was a discussion about this with hlib, and we decided to reimplement this "proxy constructor" for this reason (the alternative is changing the constructor params of state.NewCoreAccessor)

Reply:
Ah okay makes sense.